sycl/doc/extensions/experimental/sycl_ext_oneapi_non_uniform_groups.asciidoc

```diff
@@ -423,225 +424,102 @@ if (will_branch)
-
-  bool leader() const;
-};
-
-}
-----
-
-NOTE: `fixed_size_group` is templated on a `ParentGroup` because it is expected
-that it will eventually be possible to construct a fixed-size-group from more
-than only sub-groups.
-
-[source,c++]
-----
-id_type get_group_id() const;
-----
-_Returns_: An `id` representing the index of the fixed-size-group.
-
-[source,c++]
-----
-id_type get_local_id() const;
-----
-_Returns_: An `id` representing the calling work-item's position within
-the fixed-size-group.
-
-[source,c++]
-----
-range_type get_group_range() const;
-----
-_Returns_: A `range` representing the number of fixed-size-groups.
-
-[source,c++]
-----
-range_type get_local_range() const;
-----
-_Returns_: A `range` representing the number of work-items in the
-fixed-size-group, which is always equal to `PartitionSize`.
-
-[source,c++]
-----
-id_type get_group_linear_id() const;
-----
-_Returns_: A linearized version of the `id` returned by `get_group_id()`.
-
-[source,c++]
-----
-id_type get_local_linear_id() const;
-----
-_Returns_: A linearized version of the `id` returned by `get_local_linear_id()`.
-
-[source,c++]
-----
-range_type get_group_linear_range() const;
-----
-_Returns_: A linearized version of the `id` returned by `get_group_range()`.
-
-[source,c++]
-----
-range_type get_local_linear_range() const;
-----
-_Returns_: A linearized version of the `id` returned by `get_local_range()`.
-
-[source,c++]
-----
-bool leader() const;
-----
-_Returns_: `true` for exactly one work-item in the fixed-size-group, if the calling
-work-item is the leader of the fixed-size-group, and `false` for all other
-work-items in the fixed-size-group. The leader of the fixed-size-group is guaranteed
-to be the work-item for which `get_local_id()` returns 0.
-
-
-==== Usage Examples
+template <sycl::memory_order Order,
+          sycl::memory_scope Scope,
+          sycl::access::address_space AddressSpace>
+int atomic_aggregate_inc(sycl::atomic_ref<int, Order, Scope, AddressSpace> ptr) {
 
-A `fixed_size_group` can be used to apply group algorithms to subsets of data:
+  // get the set of work-items that called this function simultaneously
+  auto active_group = sycl::ext::oneapi::experimental::this_kernel::get_opportunistic_group();
 
-[source, c++]
-----
-auto sg = it.get_sub_group();
+  // increment the atomic once on behalf of all active work-items
+  int count = active_group.get_local_linear_range();
+  int old_value;
+  if (active_group.leader()) {
+    old_value = ptr.fetch_add(count);
+  }
 
-// reduce over contiguous groups of 8 elements
-auto partition = sycl::ext::oneapi::experimental::get_fixed_size_group<8>(sg);
-auto result = sycl::reduce_over_group(partition, buf[it.get_local_linear_id()], sycl::plus<>());
+  // return the value the individual work-item might have received if it had worked alone
+  auto base = sycl::group_broadcast(active_group, old_value);
+  auto idx = active_group.get_local_linear_id();
+  return base + idx;
```

Comment:
Linear id index is 0-based, meaning that for the leader work-item it would look like it haven't performed an increment at all. Should it be `linear_id + 1` here?

Reply:
I think this is correct.  Operations like `fetch_add` return the value *before* the increment happens.

A common use-case for this is that you're using the atomic to identify the first empty location in an array.  The first result should be "0", and you shouldn't see a "1" unless somebody else incremented the counter before you.

Reply:
> Operations like fetch_add return the value before the increment happens.

Oh, right, got it, thanks!